code/game/objects/structures/ladders.dm

```diff
@@ -197,3 +197,61 @@
 
 /obj/structure/ladder/crafted
 	crafted = TRUE
+
+/obj/structure/ladder/unbreakable/teleport
+	name = "mysterious ladder"
+	desc = "A mysterious ladder leading down. Where does it go?"
+
+/obj/structure/ladder/unbreakable/teleport/Initialize(mapload, obj/structure/ladder/up, obj/structure/ladder/down)
+	AddElement(/datum/element/undertile, TRAIT_T_RAY_VISIBLE, use_anchor = TRUE)
+	return ..()
+
+/obj/structure/ladder/unbreakable/teleport/LateInitialize()
+	if(down)
+		return
+
+	shuffle(GLOB.ladders)
+	for (var/obj/structure/ladder/unbreakable/teleport/ladder in GLOB.ladders)
+		if(ladder == src || ladder.down)
+			continue
+		down = ladder
+		ladder.down = src
+		ladder.update_appearance()
+		update_appearance()
+		break
+
+	if(!down)
+		qdel(src, force = TRUE)
+
+/obj/structure/ladder/unbreakable/teleport/travel(going_up, mob/user, is_ghost, obj/structure/ladder/ladder)
+	var/turf/target = get_turf(ladder)
+
+	if(!is_ghost)
```

Comment:
switch this around and early return after ghosts attempt this, you have a small else case at the end of this if proc that makes you have to indent everything